### ISSUE TITLE
FullTextTrigger optimization (proposal)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sagarswathi/h2database#84]
```
Hi!

What do you think about these changes in FullTextLucene.FullTextTrigger?

public void fire(Connection conn, Object[] oldRow, Object[] newRow) throws
SQLException {
   if (oldRow != null && newRow == null) {
      // DELETE
      delete(oldRow);
   } else if (oldRow == null && newRow != null) {
      // INSERT
      insert(newRow);
   } else if (oldRow != null && newRow != null) {
      // UPDATE
      for (int i = 0; i < indexColumns.length; i++) {
         int index = indexColumns[i];
         if (!newRow[index].equals(oldRow[index])) {
            delete(oldRow);
            insert(newRow);
            break;
         }
      }
   }
}

```

Original issue reported on code.google.com by `victor.p...@gmail.com` on 25 May 2009 at 11:17

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Thanks! I will implement this change in the next release.

Regards,
Thomas
```

Original comment by `thomas.t...@gmail.com` on 29 May 2009 at 4:08

**Comment by GoogleCodeExporter** [comment]
Original comment by `thomas.t...@gmail.com` on 29 May 2009 at 4:08
- Changed state: **Accepted**
- Added labels: **Type-Enhancement**

**Comment by GoogleCodeExporter** [comment]
Original comment by `thomas.t...@gmail.com` on 29 May 2009 at 4:08

**Comment by GoogleCodeExporter** [comment]
```
Implemented in version 1.1.114
```

Original comment by `thomas.t...@gmail.com` on 4 Jun 2009 at 7:22
- Changed state: **Fixed**
